subworkflows/local/fasta_annotation_metaprodigal_cdhit.nf

```diff
@@ -0,0 +1,45 @@
+//
+// Subworkflow for the annotation of a metagenome using Metaprodigal and CD-HIT
+//
+
+include {       PRODIGAL       } from '../../modules/nf-core/prodigal/main.nf'
+include {      CDHIT_CDHIT     } from '../../modules/nf-core/cdhit/cdhit/main.nf'
+include { FILTER as FILTER_one } from '../../modules/local/filter.nf'
+include { FILTER as FILTER_two } from '../../modules/local/filter.nf'
+
+workflow FASTA_ANNOTATION_METAPRODIGAL_CDHIT {
+    take:
+    ch_metagenome // channel in which each element is a tuple: [ val(meta), [ genome ]]
+
+    main:
+
+    // Starting empty channel to store all the softwares' versions
+    ch_versions = Channel.empty()
+
```

Comment:
check if ch_metagenome is zipped, if yes unzip else pass to filter_one directly.
https://github.com/nf-core/rnaseq/blob/master/subworkflows/local/prepare_genome.nf

Reply:
I used the filter operator:

```
    // 0. Uncompress files in compressed_files_ch channel (if necessary)
    ch_metagenome
        .filter { meta, genome_file -> genome_file.toString().endsWith('.gz') }
        .set { compressed_files_ch }

    ch_metagenome
        .filter { meta, genome_file -> !genome_file.toString().endsWith('.gz') }
        .set { uncompressed_files_ch }

    GUNZIP_one( compressed_files_ch )
    ch_versions = ch_versions.mix( GUNZIP_one.out.versions.first() )

    // 1. Filtering out contigs shorter than 500 bp (default)
    FILTER_one( GUNZIP_one.out.gunzip.mix(uncompressed_files_ch) )
    ch_versions = ch_versions.mix( FILTER_one.out.versions.first() )
```